v2/client/invoker.go

```diff
@@ -39,50 +40,55 @@ type receiveInvoker struct {
 func (r *receiveInvoker) Invoke(ctx context.Context, m binding.Message, respFn protocol.ResponseFn) (err error) {
 	var isFinished bool
 	defer func() {
+		// skip finishing if we did it explicitly already
 		if !isFinished {
-			if err2 := m.Finish(err); err2 == nil {
-				err = err2
+			if fErr := m.Finish(err); fErr == nil {
+				err = nil
 			}
 		}
 	}()
 
-	e, err := binding.ToEvent(ctx, m)
-	if err != nil {
-		return err
-	}
+	var respMsg binding.Message
+	var result protocol.Result
+
+	e, eventErr := binding.ToEvent(ctx, m)
+	switch {
+	case eventErr != nil:
+		result = protocol.NewReceipt(false, "failed to convert Message to Event: %v", eventErr)
```

Comment:
Look the comment here: https://github.com/cloudevents/sdk-go/issues/478#issuecomment-621123504

Reply:
The function is invoked with a nil message and this `result` few lines below. Did I misunderstand?

Reply:
What I meant in the comment is that, before invoking `ToEvent`, you should check the encoding of the message

Reply:
The encoding has already been checked and marked as valid when we reach here (maybe that's also an issue).

The problem occurs when fields such as `version` are missing.

Reply:
I'm sorry you're right it's checked here :smile: https://github.com/cloudevents/sdk-go/pull/480/files/ec24024f23df968a6752e321ea05324804408487#diff-255451943fd83a8be86080df4ef96fb9R221